services/restaurants/restaurant_comparer.py

```diff
@@ -0,0 +1,143 @@
+        similarity_score += temp_similarity_score
+        max_score += temp_max_score
+        print("similarity score: " + str(temp_similarity_score) + " max score: " + str(temp_max_score))
+
+        temp_similarity_score, temp_max_score = self._compare_review_count(rest1 = restInfo1, rest2 = restInfo2)
+        similarity_score += temp_similarity_score
+        max_score += temp_max_score
+        print("similarity score: " + str(temp_similarity_score) + " max score: " + str(temp_max_score))
+
+        temp_similarity_score, temp_max_score = self._compare_rating(rest1 = restInfo1, rest2 = restInfo2)
+        similarity_score += temp_similarity_score
+        max_score += temp_max_score
+        print("similarity score: " + str(temp_similarity_score) + " max score: " + str(temp_max_score))
+
+        temp_similarity_score, temp_max_score = self._compare_cuisine_type(rest1 = restInfo1, rest2 = restInfo2)
+        similarity_score += temp_similarity_score
+        max_score += temp_max_score
+        print("similarity score: " + str(temp_similarity_score) + " max score: " + str(temp_max_score))
+
+        return str(similarity_score/max_score)
+
+    def _compare_price(self, rest1: Restaurant, rest2: Restaurant):
+        max_score = 100
+        # price difference range is 0-3
+        price_difference = abs(len(rest1.price) - len(rest2.price))
+        if "" != rest1.price and rest2.price:
+            print("price 1: " + rest1.price + " price 2: " + rest2.price)
+            # return score and max score possible
+            return max_score - (33 * price_difference), max_score
+        else:
+            # if unable to factor in price, return a score of 0 and a max score of 0
+            return 0, 0
+
+    def _compare_review_count(self, rest1: Restaurant, rest2: Restaurant):
+        max_score = 20
+        
+        # ensure restaurant has been rated
+        if 0 != rest1.review_count and rest2.review_count:
+            print("review count 1: " + str(rest1.review_count) + " review count 2: " + str(rest2.review_count))
+            # review count similarity will be based on percent similarity
+            percent_similar = self._find_percent_similar(rest1.review_count, rest2.review_count)
+            return max_score * percent_similar, max_score
+
+        else:
+            return 0, 0
+    
+    def _compare_rating(self, rest1: Restaurant, rest2: Restaurant):
+        max_score = 100
+        
+        # ensure restaurant has been rated
+        if 0 != rest1.rating and rest2.rating:
+            print("rating 1: " + str(rest1.rating) + " rating 2: " + str(rest2.rating))
+            # rating similarity will be based on percent similarity
+            percent_similar = self._find_percent_similar(rest1.rating, rest2.rating)
+            return max_score * percent_similar, max_score
+
+        else:
+            return 0, 0
+
+    def _compare_cuisine_type(self, rest1: Restaurant, rest2: Restaurant):
```

Comment:
should cusine similarity score be based on the characters themselves or should it be based on location in the world to each other? Something to think about since comparing characters of the cuisine doesn't really hold any real comparison value. Compared to getting the distance between mexican and china vs mexican and american

Reply:
Distance could work, although hopefully the current algorithm does that... in a way. I forgot to add comments, but essentially wordnet finds the connotational similarity of two words. For example, "tacos" and "Mexican" have an over 90% correlation score, whereas "year" and "dear" are about 17% correlated.

I will have to test how similarly located places are correlated. But from one test, "Croatian" and "Indian" have a score of 66%, which is about right.